core/warehouse.go

```diff
@@ -0,0 +1,50 @@
+package warehouse
+
+import (
+	"errors"
+	"fmt"
+)
+
+// ErrCategoryHiearchy thrown when there is an empty category and it's not the end of the branch
+type ErrCategoryHiearchy struct {
```

Comment:
Would simply define the error as a const with the error message or the response code.